File: src/main/java/at/searles/parsing/printing/AnnotatedConcreteSyntaxTree.java
package at.searles.parsing.printing;

public class AnnotatedConcreteSyntaxTree<C> implements ConcreteSyntaxTree {
    private final C annotation;
    private final ConcreteSyntaxTree child;

    public AnnotatedConcreteSyntaxTree(C annotation, ConcreteSyntaxTree child) {
        this.child = child;
        this.annotation = annotation;
    }

    public String toString() {
        return child.toString();
    }

    @Override
    public void printTo(CstPrinter printer) {
        printer.print(child, annotation);
    }
}

File: src/main/java/at/searles/parsing/printing/AnnotatedConcreteSyntaxTree.kt
package at.searles.parsing.printing

class AnnotatedConcreteSyntaxTree<C>(private val annotation: C, private val child: ConcreteSyntaxTree) : ConcreteSyntaxTree {
    override fun toString(): String {
        return child.toString()
    }

    override fun printTo(printer: CstPrinter) {
        printer.print(child, annotation)
    }

}